cli/proc_state.rs

```diff
@@ -412,6 +425,14 @@ impl ProcState {
         .await?;
       self.prepare_node_std_graph().await?;
     }
+    // If there's a lock file, update it with all discovered npm packages
+    if let Some(lockfile_mutex) = &self.lockfile {
+      let mut lockfile = lockfile_mutex.lock();
+      if let Err(err) = self.npm_resolver.lock(&mut lockfile) {
+        log::error!("{} {}", colors::red("error:"), err);
+        std::process::exit(10);
```

Comment:
We probably shouldn't exit the process here because this code might be used in the lsp.

Reply:
How would you suggest to handle that then?

Reply:
This actually links to https://github.com/denoland/deno/pull/15938#discussion_r1004468461 - if we do that for each package then it will kill the LSP.

Reply:
Maybe we can improve `unwrap_or_exit` to handle an error type that includes an exit code